packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -600,6 +618,19 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
       ? props.clusterIdentifier?.toLowerCase()
       : props.clusterIdentifier;
 
+    if (props.domain) {
+      this.domainId = props.domain;
+      this.domainRole = props.domainRole ?? new iam.Role(this, 'RDSClusterDirectoryServiceRole', {
+        assumedBy: new iam.CompositePrincipal(
+          new iam.ServicePrincipal('rds.amazonaws.com'),
+          new iam.ServicePrincipal('directoryservice.rds.amazonaws.com'),
```

Comment:
In the implementation of the instance, `directoryservice.rds.amazonaws.com` is not set as a principal, which is a bug and causes the deployment to fail.

We will address this in a separate issue(#28600).

Reply:
Nice catch!